lumicks/pylake/tests/test_imaging_mixins.py

```diff
@@ -96,47 +97,63 @@ def test_export_tiff_empty(tmp_path):
-    images = np.ones(shape=(2, 10, 10, 3))  # (n, h, w, c)
-    export_image16 = tiffexport_factory(images * np.iinfo(np.uint16).max)
-
-    # Sufficient bit-depth or forced clipping
-    export_image16.export_tiff(tmp_path / "uint16", dtype=np.uint16)
-    export_image16.export_tiff(tmp_path / "float32", dtype=np.float32)
-    export_image16.export_tiff(tmp_path / "clipped", dtype=np.uint8, clip=True)
-
-    # Raise because unsafe
-    with pytest.raises(RuntimeError, match="Can't safely export image with `dtype=uint8` channels"):
-        export_image16.export_tiff(tmp_path / "uint8", dtype=np.uint8)
+@pytest.mark.parametrize(
+    "output_dtype, data_dtype",
+    [
+        (np.float32, np.float32),
+        (np.float32, np.float16),
+        (np.float32, np.uint32),
+        (np.float16, np.float16),
+        (np.uint32, np.uint16),
+        (np.uint32, np.uint8),
+        (np.uint16, np.uint16),
+        (np.uint16, np.uint8),
+    ],
+)
+def test_export_tiff_sufficient_range(tmp_path, output_dtype, data_dtype):
+    """Test TIFF export when the data type has sufficient range for the data stored"""
+    info = np.finfo(data_dtype) if np.dtype(data_dtype).kind == "f" else np.iinfo(data_dtype)
+    images = np.ones(shape=(2, 10, 10, 3)) * info.max  # (n, h, w, c)
+    export_image = tiffexport_factory(images)
+    export_image.export_tiff(tmp_path / "tmp", dtype=output_dtype)
 
-    with pytest.raises(
-        RuntimeError, match="Can't safely export image with `dtype=float16` channels"
-    ):
-        export_image16.export_tiff(tmp_path / "float16", dtype=np.float16)
+    # Validate data
+    with TiffFile(tmp_path / "tmp") as t:
+        np.testing.assert_allclose(t.asarray(), images)
 
 
-@pytest.mark.filterwarnings(
-    # Numpy 1.24 raises a RuntimeWarning upon overflow during type cast (see call with `clip=True`)
-    "ignore:overflow encountered in cast:RuntimeWarning:lumicks.pylake.detail.imaging_mixins:48"
+@pytest.mark.parametrize(
+    "output_dtype, data_dtype",
+    [
+        (np.float16, np.float32),
+        (np.uint8, np.float32),
+        (np.uint16, np.float32),
+        (np.uint32, np.float32),
+        (np.uint16, np.uint32),
+        (np.uint8, np.uint16),
+        (np.uint8, np.uint32),
+    ],
 )
-def test_export_tiff_float(tmp_path):
-    images = np.ones(shape=(2, 10, 10, 3))  # (n, h, w, c)
-    export_image32 = tiffexport_factory(images * np.finfo(np.float32).max)
-
-    # Sufficient bit-depth or forced clipping
-    export_image32.export_tiff(tmp_path / "float32", dtype=np.float32)
-    export_image32.export_tiff(tmp_path / "clipped", dtype=np.float16, clip=True)
+def test_export_tiff_insufficient_range(tmp_path, output_dtype, data_dtype):
+    """Test TIFF export when the data type has insufficient range for the data stored"""
+    info = np.finfo(data_dtype) if np.dtype(data_dtype).kind == "f" else np.iinfo(data_dtype)
+    images = np.ones(shape=(2, 10, 10, 3)) * info.max  # (n, h, w, c)
```

Comment:
Probably separate/parameterize test into `info.min` and `info.max`.

Reply:
Good call! Parameterized it.